u2-routing-controllers/app.js

```diff
@@ -0,0 +1,31 @@
+    const express = require('express')
+    const { urlencoded, json } = require ('body-parser')
+    const { getUserByQuery, getUserByPath, getUserByForm, search } = require('./features/userController')
+    const { logger } = require('./_services/logger')
+    const { index } = require('./features/indexController')
+    const { getproduct } = require('./features/productController')
+    const { getRGB } = require('./features/rgbController')
+    const { getTemp } = require('./features/tempController')
+    const { getEcho, getEchoByForm } = require('./features/echoController')
+
+    const app = express()
+
+    //Middleware
+    app.use(urlencoded({ extended: false }))
+    app.use(json())
+    app.use(logger)
+
+    // Routes
+    app.get('/',index)
+    app.get('/users', getUserByQuery)
+    app.get('/users/search', search)
+    app.get('/users/:username', getUserByPath)
```

Comment:
ด้วย ExpressJS (เฟรมเวิร์กเว็บฝั่งเซิร์ฟเวอร์ที่เราเลือก) จำเป็นต้องวาง `/users/search` ไว้เหนือ `/users/:username` (เหมือนที่คุณทำ) หากคุณใส่ไว้ด้านล่าง ผู้ใช้ที่ขอเส้นทางการค้นหาจะตรงกับเส้นทางโปรไฟล์ผู้ใช้ก่อน! งานดีที่นี่ :)

With ExpressJS (our chosen server-side web framework), it's important to put `/users/search` above `/users/:username` (like you did). If you put it below, a user requesting the search route would match the user profile route first! Good job here :)